dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,327 @@
+                        device_info["Name"] = self._extract_network_device_name(net_cfg_instance_id)
+
+                        device_info["NetworkAddress"] = self._try_value(subkey, "NetworkAddress")
+                        device_info["NetworkInterfaceInstallTimestamp"] = wintimestamp(
+                            self._try_value(subkey, "NetworkInterfaceInstallTimestamp")
+                        )
+                        device_info["DriverDesc"] = self._try_value(subkey, "DriverDesc")
+                        device_info["Type"] = IfTypes(int(self._try_value(subkey, "*IfType"))).name
+                        device_info["Vlan"] = PriorityVlanTagEnum.get(
+                            int(self._try_value(subkey, "*PriorityVlanTag")), "OTHER"
+                        )
+
+                        if "Type" in device_info:
+                            yield WindowsInterfaceRecord(
+                                _target=self.target,
+                                dns=device_info["dns"],
+                                search_domain=device_info["domain"],
+                                gateway=device_info["gateway"],
+                                ip=device_info["ip"],
+                                first_connected=device_info["NetworkInterfaceInstallTimestamp"],
+                                mac=device_info["NetworkAddress"],
+                                metric=device_info["metric"],
+                                name=device_info["Name"],
+                                source="registry",
+                                network=device_info["subnetmask"],
+                                type=device_info["Type"],
+                                vlan=device_info["Vlan"],
+                                last_connected=None,
+                                enabled=None,
+                            )
+
+    def _extract_network_device_config(self, interface_id) -> list[dict, str]:
+        config = {"domain": "", "gateway": [], "ip": [], "dns": [], "subnetmask": [], "metric": 0}
+
+        try:
+            keys = self.target.registry.key(REGISTRY_KEY_INTERFACE + interface_id)
+        except RegistryKeyNotFoundError:
+            return
+
+        if not len(keys):
+            return
+
+        for config_item in list(keys.values()):
+            name = config_item.name
+            value = config_item.value
+            if name in MAPDEF and value not in ["", "0.0.0.0", 0]:
+                if isinstance(value, str):
+                    if name in ["DhcpDomain", "Domain"]:
+                        config[MAPDEF[name]] = str(value)
+                    else:
+                        config[MAPDEF[name]].extend([value])
+                elif isinstance(value, list):
+                    config[MAPDEF[name]].extend(value)
+                elif isinstance(value, int):
+                    config[MAPDEF[name]] = value
+
+        if any(config.values()):
+            return config
```

Comment:
This function is very hard to comprehend and understand what it actually does, while what it's supposed to do is actually very simple. I'd lose the dictionary stuff and looping stuff and just make it a static series of lookups, value extractions and assignments over the individual values.